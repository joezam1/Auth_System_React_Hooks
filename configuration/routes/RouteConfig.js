const routeConfiguration = Object.freeze({
    homePath:'/',
    authLoginPath:'/auth/login',
    authRegisterPath:'/auth/register',
    authLogoutPath:'/auth/logout',
    privateCustomerDashboardPath:'/private/customer/dashboard',
    privateCustomerOrdersPath:'/private/customer/orders',
    privateCustomerSilverOffersPath:'/private/customer/silverOffers',
    privateCustomerGoldOffersPath:'/private/customer/goldOffers',
    privateCustomerPremiumOffersPath:'/private/customer/premiumOffers',
    privateCustomerPremiumClubMemberPath:'/private/customer/premiumClubMember',
    privateCustomerNewsletterPath:'/private/customer/newsletter',
    privateCustomerManagerOrdersPath:'/private/customer/ManagerOrders',
    privateCustomerContactSuppliersPath:'/private/customer/contact/suppliers',
    storeOrdersPath:'/store/orders',
    storeInvoicesPath:'/store/invoices',
    storeWarehousePath:'/store/warehouse',
    storeAdvertisingPath:'/store/advertising'

});

export default routeConfiguration;