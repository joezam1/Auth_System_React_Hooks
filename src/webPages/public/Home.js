import React from 'react';
import {Link } from 'react-router-dom';
import RouteConfig from '../../../configuration/routes/RouteConfig.js';

export default function Home(){
  return(
      <div className="home">
          This is home Section - Public Page
          <ul>
              <li> <Link to={RouteConfig.authRegisterPath} data-testid="link-home-register-id">Register</Link></li>
              <li> <Link to={RouteConfig.authLoginPath} data-testid="link-home-login-id"> Login</Link></li>
          </ul>
      </div>
  );
}