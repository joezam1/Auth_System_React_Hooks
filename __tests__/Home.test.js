import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render, cleanup, fireEvent } from '@testing-library/react';
import Home from '../src/webPages/public/Home.js';
import RouteConfig from '../configuration/routes/RouteConfig.js';


describe('File: Home.js', function(){
    afterEach(cleanup);
    describe('Component Home', function(){
        let container;
        beforeEach(()=>{
            container = render( <MemoryRouter> <Home/>  </MemoryRouter>  );
        });

        test('It should take a snapshot', function(){
            expect(container.asFragment(<Home/>)).toMatchSnapshot();
        });


        test('Register Link Redirects Correctly', function(){
                const link = container.getByText( 'Register' );
                const link1 = container.getByTestId('link-home-register-id');
                fireEvent.click(link);
                var initialPath = window.location.pathname;
                var pathname = link.pathname;
                expect(pathname).toBe(RouteConfig.authRegisterPath);
                expect(link).toBe(link1);
        });

        test('Login Link Redirects Correctly', function(){
            const link = container.getByText( 'Login' );
            const link1 = container.getByTestId('link-home-login-id');
            fireEvent.click(link);
            var initialPath = window.location.pathname;
            var pathname = link.pathname;
            expect(pathname).toBe(RouteConfig.authLoginPath);
            expect(link).toBe(link1);
    });

    });
});