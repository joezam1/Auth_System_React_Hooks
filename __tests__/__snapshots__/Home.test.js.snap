// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File: Home.js Component Home It should take a snapshot 1`] = `
<DocumentFragment>
   
  <div
    class="home"
  >
    This is home Section - Public Page
    <ul>
      <li>
         
        <a
          data-testid="link-home-register-id"
          href="/auth/register"
        >
          Register
        </a>
      </li>
      <li>
         
        <a
          data-testid="link-home-login-id"
          href="/auth/login"
        >
           Login
        </a>
      </li>
    </ul>
  </div>
    
</DocumentFragment>
`;
